docs/src/main/asciidoc/security.adoc

```diff
@@ -96,12 +98,32 @@ Additionally it provides `JWT Generation API` for creating `signed`, `inner-sign
 
 See the link:security-jwt[Using SmallRye JWT] guide for more information.
 
-=== OAuth2
+==== OAuth2
 
 `quarkus-elytron-security-oauth2` provides an alternative to `quarkus-oidc` Bearer Token Authentication Mechanism. It is based on `Elytron` and is primarily meant for introspecting the opaque tokens remotely.
 
 See the link:security-oauth2[Using OAuth2] guide for more information.
 
+==== Choosing between OpenId Connect, SmallRye JWT and OAuth2 extensions
+
+`quarkus-oidc` extension requires an OpenId Connect provider such as Keycloak which can be used to verify the Bearer tokens or authenticate the human users with the Authorization Code flow. In both cases `quarkus-oidc` requires a connection to this OpenId Connect provider.
+
+`quarkus-oidc` is the only option when the user authentication via Authorization Code flow  or supporting multiple tenants is required.
```

Comment:
```suggestion
`quarkus-oidc` is the only option when the user authentication via Authorization Code flow or supporting multiple tenants is required.
```

Reply:
@gastaldi sorry, resolved by accident :-).
I was thinking of coming up with a more opinionated recommendation on which extension to use, it is not there yet, but I can try a table as a summary and then follow up with some text, and I'll experiment :-)